backend/app/app.go

```diff
@@ -128,7 +128,7 @@ func NewApp(config internal.Configuration) (*App, error) {
 	}
 	sshPublicKey := strings.TrimSpace(string(sshPublicKeyBytes))
 
-	_, appCancel := context.WithCancel(context.Background())
+	_, appCancel := context.WithCancel(ctx)
```

Comment:
I had a memory of the cancel function being stored on the app, if not we can store this one and call it during the exit